### PR TITLE
ysld generate styles break kml output formats [GEOS-8885]

### PR DESCRIPTION
**Pull request by jodygarnett**
See [GEOS-8885](https://osgeo-org.atlassian.net/browse/GEOS-8885) for details.

Will need to check for code that assumes expressions are provided:

* [x] PlacemarkNameDecoratorFactory: label used if available
* [x] PlacemarkStyleDecorator PolygonSymbolizer fill-color, fill-opacity
* [x] PlacemarkStyleDecorator LineSymbolizer: stroke-opacity, stroke-color, stroke-width
* [x] PlacemarkStyleDecorator TextSymbolizer:  fill-opacity, fill-color, font-size
* [x] IconPropertyExtractor: mostly has null checks in place

### CODE REVIEW

File: src/kml/src/test/java/org/geoserver/kml/icons/IconPropertiesTest.java [hunk 1]
```diff
@@ -83,6 +85,20 @@ public void testDynamicMark() throws CQLException {
...
+        Expression color = toExpression("if_then_else(equalTo(field, 1), '#8080C0', '#CC8030')");
+        Stroke stroke = styleFactory.stroke(color,  null,  null, null, null, null, null);
+        Fill fill = styleFactory.fill(null, color, null );        
+        Mark mark = styleFactory.mark(toExpression("circle"), fill, stroke);
+        Graphic graphic = styleFactory.graphic( Collections.singletonList(mark),null, null, null, null, null);       
+        PointSymbolizer symbolizer = styleFactory.pointSymbolizer("symbolizer", toExpression("geom"),  null,  null, graphic );
+        
+        final Style s = styleFromRules(catchAllRule(symbolizer));
+        assertEquals("0.0.0=&0.0.0.fill.color=%238080C0&0.0.0.name=&0.0.0.stroke.color=%238080C0", encode(s, fieldIs1));
+        assertEquals("0.0.0=&0.0.0.fill.color=C08030&0.0.0.name=&0.0.0.stroke.color=C08030", encode(s, fieldIs2));
```
Comment (jodygarnett) [inline src/kml/src/test/java/org/geoserver/kml/icons/IconPropertiesTest.java]: The above encoding shows RGB hex, expect we should be using ABGR order hex.

Comment (aaime) [reply src/kml/src/test/java/org/geoserver/kml/icons/IconPropertiesTest.java]: Hmm... don't know honestly? But it's failing:

{code}
Failed tests:   testDynamicColor(org.geoserver.kml.icons.IconPropertiesTest): expected:<...0=&0.0.0.fill.color=[C08030&0.0.0.name=&0.0.0.stroke.color=C0]8030> but was:<...0=&0.0.0.fill.color=[%23CC8030&0.0.0.name=&0.0.0.stroke.color=%23CC]8030>
{code}

Check the icon server expectations (was not written by me)

**Comment by aaime** [comment]
Looks good

**Comment by jodygarnett** [comment]
As this is a bug fix I would like to back port, squashed into a single commit to make this easier.

**Comment by aaime** [comment]
As a bug fix you can do so without any discussion :)

**Pull request merged**